test/PandoraInterface.cxx

```diff
@@ -163,36 +160,36 @@ void CreateGeometry(const Parameters &parameters, const Pandora *const pPrimaryP
     pCurrentShape->InspectShape();
     TGeoBBox *pBox = dynamic_cast<TGeoBBox *>(pCurrentShape);
 
-    // mm to cm conversion
-    const double mm2cm(0.1);
-
     // Now can get origin/width data from the BBox
-    double dx = pBox->GetDX() * mm2cm; // Note these are the half widths
-    double dy = pBox->GetDY() * mm2cm;
-    double dz = pBox->GetDZ() * mm2cm;
+    const float dx = pBox->GetDX() * parameters.m_mm2cm; // Note these are the half widths
+    const float dy = pBox->GetDY() * parameters.m_mm2cm;
+    const float dz = pBox->GetDZ() * parameters.m_mm2cm;
     const double *origin = pBox->GetOrigin();
 
     // Translate the origin coordinates from the 4th level to the first.
-    // Doesn't seem to change anything. Needed?
-    Double_t level3[3] = {0.0, 0.0, 0.0};
+    // TODO: Check if this is really needed, as it doesn't seem to change anything
+    double level3[3] = {0.0, 0.0, 0.0};
     currentnode->LocalToMasterVect(origin, level3);
-    Double_t level2[3] = {0.0, 0.0, 0.0};
+    double level2[3] = {0.0, 0.0, 0.0};
     currentnode->LocalToMasterVect(level3, level2);
-    Double_t level1[3] = {0.0, 0.0, 0.0};
+    double level1[3] = {0.0, 0.0, 0.0};
```

Comment:
Is the idea here that the local and intermediate coordinates here come from boxes, within boxes, within boxes, and that these steps promote the local coordinates to the global coordinates, or is something other structure involved here? That these translations apparently don't do anything suggest I have the wrong model.

Reply:
We only need 1 local-to-global transform so the code has been updated accordingly.